ledger-api/grpc-definitions/com/daml/ledger/api/v1/admin/user_management_service.proto

```diff
@@ -0,0 +1,175 @@
+    // _provided_ it has the corresponding ``CanReadAs(primary_party)`` or 
+    // ``CanActAs(primary_party)`` rights.
+    string primary_party = 2;
+}
+
+// A right granted to a user.
+message Right {
+    // The user is allowed to administrate the participant node.
+    message ParticipantAdmin {}
+
+    // The user can authorize commands for the given party.
+    message CanActAs {
+        string party = 1;
+    }
+
+    // The user can read ledger data visible to the given party.
+    message CanReadAs {
+        string party = 1;
+    }
+
+    oneof kind {
+        ParticipantAdmin participant_admin = 1;
+        CanActAs can_act_as = 2;
+        CanReadAs can_read_as = 3;
+    }
+}
+
+
+// RPC requests and responses
+/////////////////////////////
+
+// Required authorization: ``HasRight(ParticipantAdmin)``
+message CreateUserRequest {
+    // The user to create.
+    User user = 1;
+
+    // The rights to be assigned to the user upon creation,
+    // which SHOULD include appropriate rights for the ``user.primary_party``.
+    repeated Right rights = 2;
+}
+
+// Required authorization: ``HasRight(ParticipantAdmin) OR IsAuthenticatedUser(user_id)``
+message GetUserRequest {
+    // The user whose data we want to retrieve.
+    // If set to empty string (the default), then the data for the authenticated user will be retrieved.
+    string user_id = 1;
+}
+
+// Required authorization: ``HasRight(ParticipantAdmin)``
+message DeleteUserRequest {
+    string user_id = 1;
+}
+
+// Required authorization: ``HasRight(ParticipantAdmin)``
+message ListUsersRequest {
+    // TODO: add pagination, cf. https://cloud.google.com/apis/design/design_patterns#list_pagination
+}
+
+message ListUsersResponse {
+    repeated User users = 1;
+}
+
+// Add the rights to the set of rights granted to the user.
+//
+// Required authorization: ``HasRight(ParticipantAdmin)``
+message GrantUserRightsRequest {
+    string user_id = 1;
+
+    repeated Right rights = 2;
+}
+
+message GrantUserRightsResponse {
+    // The rights that were newly granted by the request.
+    repeated Right newly_granted_rights = 1;
+}
+
+// Remove the rights from the set of rights granted to the user.
+//
+// Required authorization: ``HasRight(ParticipantAdmin)``
+message RevokeUserRightsRequest {
+    string user_id = 1;
+
+    repeated Right rights = 2;
+}
+
+message RevokeUserRightsResponse {
+    // The rights that were actually revoked by the request.
+    repeated Right newly_revoked_rights = 1;
+}
+
+// Required authorization: ``HasRight(ParticipantAdmin) OR IsAuthenticatedUser(user_id)``
+message ListUserRightsRequest {
+    // The user for which to list the rights.
+    // If set to empty string (the default), then the rights for the authenticated user will be listed.
+    string user_id = 1;
+
+    // TODO: add pagination, cf. https://cloud.google.com/apis/design/design_patterns#list_pagination
+}
+
+message ListUserRightsResponse {
+    repeated Right rights = 1;
```

Comment:
In Daml script, the natural thing to do seems to be to flip this around immediately and turn it into:
```
data Rights = Rights
  { actAs : [Party]
  , readAs : [Party]
   , admin : Bool
   }
```
I could see other clients wanting to do the same thing (the JSON API for example). Have you considered emitting the result in something closer to that format?

Reply:
I have tried it, and it was not easy to present that way. The flat set of rights is a simple, well-defined structure that lends itself well to extensions.